lib/linguist/heuristics.rb

```diff
@@ -541,5 +541,13 @@ def call(data)
       end
     end
 
+    disambiguate ".yy" do |data|
+      Language["JSON"] if /\"modelName\"\:\s?\"/.match(data)
+    end
+
+    disambiguate ".yyp" do |data|
+      Language["JSON"] if /\"modelName\"\:\s?\"GMProject\"/.match(data)
+    end
```

Comment:
Is this required? Did you find other languages using `.yyp`?

Reply:
Just curious: is there a guarantee that only one or zero whitespace characters follow `modelName:`? If not, you should replace `\s?` with `\s*` so it'll match 2+ blanks instead of 1 or none. 😉

Reply:
Good one, the * would be better. They are automatically generated files though, so I'm sure its usually 1 whitespace, but the * is better. Committing as we speak!

Reply:
I've extended it since I've checked pretty much all .yy files, and they all contain a modelName element with a value that starts with ``GM``.

``"modelName": "GM`` a match on that one is what I'm doing for this rule.